common/script/index.coffee

```diff
@@ -1695,10 +1695,93 @@ api.wrap = (user, main=true) ->
       {user}
     ###
     cron: (options={}) ->
+      # console.log("\n\n------ cron -------")
       now = +options.now || +new Date
-
-      daysMissed = api.daysSince user.lastCron, _.defaults({now}, user.preferences)
+      # console.log {now:moment(now).format('YYYY-MM-DD HH:mm:ss')}
+
+      # If the user's timezone has changed (due to travel or daylight savings),
+      # cron can be triggered twice in one day. So if a zone change is detected,
+      # we ASSUME cron might have already run today and prevent some parts of
+      # a second cron (e.g., prevent damage from missed Dailies).
+      # FIXME: This can probably also prevent the day's first cron under some
+      # circumstances, and in those cases all cron code should be run.
+      userTimezoneOffset = +user.preferences.timezoneOffset || 0
+      browserTimezoneOffset = +options.timezoneOffset || userTimezoneOffset
+      # console.log {userTimezoneOffset}
+      # console.log {browserTimezoneOffset}
+
+      # This considers the case where the timezone is changing so that time
+      # goes backwards (e.g., at 2am the clocks change to 1am). More work is
+      # needed for the opposite direction.
+
+      # How many days have we missed using the timezone stored in the user's Habitica preferences:
+      daysMissedOldZone = api.daysSince user.lastCron, _.defaults({now}, user.preferences)
+      # console.log {daysMissedOldZone:daysMissedOldZone}
+
+      daysMissed = 0
+      if userTimezoneOffset == browserTimezoneOffset
+        # No timezone change; normal cron processing.
+        daysMissed = daysMissedOldZone
+        # console.log("\nsame zones:")
+      else
+        # console.log("\ndifferent zones:")
+        # console.log {userTimezoneOffset}
+
+        endOfDay = api.startOfDay(_.defaults {now}, user.preferences)
+        endOfDay = moment(endOfDay).add({days:1})
+        endOfDay = endOfDay.subtract({minutes:1})
```

Comment:
This endOfDay value needs to be assigned to lastCron under some circumstances to prevent any chance of a cron running before the user's next CDS. But I can't get that darned value right. I'm sure it's a simple fix that will make me want to scream when I see it.